common/xdc/eventReplication.go

```diff
@@ -0,0 +1,128 @@
+package xdc
+
+import (
+	"github.com/uber/cadence/common"
+)
+
+const (
+	emptyCommonAncestor = -1
+)
+type (
+	eventNode struct {
+		eventID int64
+		version int64
+	}
+
+	eventBranch struct {
+		branchToken []byte
+		versionHistory []eventNode
+	}
+
+	eventReplicationMetadata struct {
+		eventsBranches [] eventBranch
+	}
+)
+
+//Not thread-safe
+func newEventBranch(token []byte, eventNodes ...eventNode) *eventBranch {
+	return &eventBranch{
+		branchToken: token,
+		versionHistory: eventNodes,
+	}
+}
+
+// add is to add new node into the versionHistory slice
+func (e *eventBranch) add(node eventNode) {
+	if node.eventID <= e.getLastNode().eventID || node.version <= e.getLastNode().version {
+		// Or panic?
+		return
+	}
+	e.versionHistory = append(e.versionHistory, node)
+}
+
+// updateEventID updates the last event id
+func (e *eventBranch) updateEventID(eventID int64) {
+	if eventID <= e.getLastNode().eventID {
+		// Or panic?
+		return
+	}
+	e.getLastNode().eventID = eventID
+}
+
+// append is to add or update the versionHistory slice
+func (e *eventBranch) append(node eventNode) {
```

Comment:
this function can be merged with updateEventID and add

Reply:
Merge those three func to one func